src/ts/cmd/endpoint-registry-ui/i40-aas-registry-ui/webapp/controller/CreateDescriptor.controller.js

```diff
@@ -0,0 +1,313 @@
+
+
+sap.ui.define([
+  "sap/ui/core/mvc/Controller",
+  "sap/ui/core/routing/History",
+  "sap/ui/model/json/JSONModel",
+  "sap/m/MessageToast",
+  "sap/m/MessageBox",
+
+
+], function (Controller, History, JSONModel, MessageToast, MsgBox) {
+  "use strict";
+
+  return Controller.extend("i40-aas-registry-ui.i40-aas-registry-ui.controller.CreateDescriptor", {
+
+    onInit: function () {
+
+      this.initiateModel();
+    },
+
+    getInputs: function () {
+      return {
+        inputAasId: this.getView().byId("InputAasId"),
+        inputAssetId: this.getView().byId("InputAssetId"),
+        createButton: this.getView().byId("CreateButton"),
+        endpointAddress: this.getView().byId("EndpointAddress"),
+      }
+    },
+
+    initiateModel: function () {
+
+      // Use Object Lib for IdType Dropdown menu
+      var oIdTypes = aas.IdTypeEnum
+      var IdTypeKeys = Object.keys(oIdTypes);
+
+      var aIdTypes = new Array();
+      for (var i = 0; i < IdTypeKeys.length; i++) {
+        var oObject = {};
+        oObject["TypeId"] = IdTypeKeys[i];
+        oObject["Name"] = IdTypeKeys[i];
+				aIdTypes.push(JSON.parse(JSON.stringify(oObject)));
+			}
+
+
+      var aEndpointTypes = (function () {
+        var aEndpointTypes = null;
+        $.ajax({
```

Comment:
While using jQuery, try to use jQuery instead of $ since it can result in some conflicts with other libraries.
SAPUI5 does not recommend to use jQuery anymore since they are planning to remove it as a hard dependency.
Please explore the following link

https://sapui5.hana.ondemand.com/#/topic/a075ed88ef324261bca41813a6ac4a1c